### DISCUSSION TITLE
反编译时遇到的一个C# 带参数属性的疑问

### DISCUSSION THREAD

**Discussion by yang0biao** [BYJRK/DotNet-Discussions#35] [Q&A]
我二次开发别的软件时，反编译了一个接口IGH_Structure，它有一个属性是有参数
```C#
/// <summary>
/// Gets the data path at the specified index.
/// </summary>
/// <param name="index">Index of path to retrieve.</param>
/// <exception cref="T:System.IndexOutOfRangeException">Thrown if the index does not represent a valid path.</exception>
GH_Path Path { get; }
```
这个接口的作者在GH_Structure类中对这个属性的实现是这样的
```C#
public GH_Path Path => m_data.Keys[index];
```

疑问一：当我使用vs实现这个接口的时候，vs并没有给我自动创建这个属性，而是生成了一个`get_Path(int index)`方法
疑问二：并且实例化这个GH_Structure类，无法访问这个属性，而只能访问到这个get_Path方法

这是什么特殊的语法吗？

**Comment by KiyanYang** [comment]
猜你在找 https://developer.rhino3d.com/api/grasshopper/html/P_Grasshopper_Kernel_Data_GH_Structure_1_Path.htm

属性是无参的，文档 https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/classes#1473-accessors

你应该想找的是索引器 https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/classes#149-indexers

**Reply by yang0biao** [reply]
应该不是索引器，GH_Structure索引器在别的地方写着。
```C#
[IndexerName("DataList")]
public List<T> this[int index] => m_data.Values[index];

[IndexerName("DataList")]
public List<T> this[GH_Path path]
{
    ....
}
```
如果这个地方不是什么我不知道的语法的话，我猜测他是反编译出错了：就是这个接口原本有一个`get_Path(int index)`方法，反编译器把它当成属性的get方法了。不知道这个猜测对不对？
